HousingSearchApi/V1/Gateways/SearchPersonESHelper.cs

```diff
@@ -0,0 +1,114 @@
+        }
+
+        private QueryContainer SearchPreferredFirstnames(GetPersonListRequest request, QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchPreferredFirstnames
+                = q.Wildcard(m =>
+                    m.Field(f => f.PreferredFirstname).Value($"*{searchText}*"));
+
+            return searchPreferredFirstnames;
+        }
+
+        private QueryContainer SearchPreferredSurnames(GetPersonListRequest request, QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchPreferredSurnames
+                = q.Wildcard(m =>
+                    m.Field(f => f.PreferredSurname).Value($"*{searchText}*"));
+
+            return searchPreferredSurnames
+                ;
+        }
+
+        private QueryContainer SearchFirstNames(GetPersonListRequest request,
+            QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchFirstNames = q.Wildcard(m =>
+                m.Field(f => f.Firstname).Value($"*{searchText}*"));
+
+            return searchFirstNames;
+        }
+
+        private QueryContainer SearchLastNames(GetPersonListRequest request,
+            QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchLastNames = q.Wildcard(m =>
+                m.Field(f => f.Surname).Value($"*{searchText}*"));
+
+            return searchLastNames;
+        }
+
+        private QueryContainer SearchMiddleNames(GetPersonListRequest request,
+            QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchMiddleNames = q.Wildcard(m =>
+                m.Field(f => f.MiddleName).Value($"*{searchText}*"));
+
+            return searchMiddleNames
+                ;
+        }
+
+        private QueryContainer SearchDateOfBirth(GetPersonListRequest request,
+            QueryContainerDescriptor<QueryablePerson> q)
+        {
+            if (string.IsNullOrWhiteSpace(request.SearchText)) return null;
+            var searchText = request.SearchText?.Replace(" ", "").ToLower();
+
+            var searchDoB = q.Wildcard(m =>
+                m.Field(f => f.DateOfBirth).Value($"*{searchText}*"));
```

Comment:
All of these private functions are identical but for the field name specified on the Wildcard. Is there any way this could be made more generic and the field required passed in? (May well not be, but worth asking...)

Reply:
Definitely. I want to do the same thing, and actually move them into individual interfaces,then inject them to one another (and thus unit test them in isolation ! )
I copied them over from addressApi just as they were for now (which was, private classes) and will improve in a future iteration. Possibly when I do pagination. 100% agree :)